### ISSUE TITLE
Cannot build GSL

### ISSUE THREAD

**Issue by stakaz** [JuliaMath/GSL.jl#70]
Hello, I canot build the package with the following error:

```julia
julia> Pkg.build("GSL")
INFO: Building GSL
Installing dependency gsl via `sudo pacman -S --needed gsl`:
warning: gsl-2.5-1 is up to date -- skipping
 there is nothing to do
=====================================[ ERROR: GSL ]=====================================

LoadError: Provider BinDeps.PackageManager failed to satisfy dependency libgsl
while loading /home/gluon/.julia/v0.6/GSL/deps/build.jl, in expression starting on line 30
```

gsl is installed as it can be seen from the message. What is the issue here?

I am on julia 0.6.2.

**Comment by jagot** [comment]
I experience the same issue on Arch Linux. By manually creating a `deps.jl` file pointing to `/usr/lib/libgsl.so` I am able to get a little bit further, but end up with the following error message:

```
julia> using GSL
INFO: Precompiling module GSL.
ERROR: LoadError: error compiling __init__: could not load library "/usr/lib/libgsl.so"
/usr/lib/libgsl.so: undefined symbol: cblas_ctrmv
Stacktrace:
 [1] anonymous at ./<missing>:2
while loading /home/jagot/.julia/v0.6/GSL/src/GSL.jl, in expression starting on line 26
```
Possibly related to this: https://github.com/dealii/dealii/issues/2757#issuecomment-230905648

**Comment by jagot** [comment]
Can confirm that commenting out package manager-related providers from `build.jl` works, forcing a build from source.

**Comment by mikolajpp** [comment]
Same thing here, also archlinux.

**Comment by simonbyrne** [comment]
We should move this to BinaryBuilder: anyone care to take this on?

**Comment by toftul** [comment]
The same on Manjaro machine with Julia 1.0. Any solutions?

**Comment by AshtonSBradley** [comment]
On macOS I get (julia 1.0)
```julia
(v1.0) pkg> build GSL
  Building LibCURL ─→ `~/.julia/packages/LibCURL/Z32eb/deps/build.log`
  Building WinRPM ──→ `~/.julia/packages/WinRPM/Y9QdZ/deps/build.log`
  Building Homebrew → `~/.julia/packages/Homebrew/l8kUw/deps/build.log`
┌ Error: Error building `Homebrew`, showing the last 100 of log: 
│ jhipster
│ jsonrpc-glib
│ kakoune
│ lf
│ libdazzle
│ libsoup
│ libxlsxwriter
│ mercurial
│ mesos
│ metashell
│ mmseqs2
│ mongodb
│ multimarkdown
│ nats-streaming-server
│ nghttp2
│ nsq
│ nss
│ oniguruma
│ openrtsp
│ pdftoedn
│ pdns
│ pdnsrec
│ php-cs-fixer
│ phpunit
│ pygobject3
│ quicktype
│ rclone
│ shibboleth-sp
│ skaffold
│ sphinx-doc
│ swiftformat
│ template-glib
│ tile38
│ topgrade
│ travis
│ uftp
│ unrar
│ v8
│ vte3
│ webpack
│ wildfly-as
│ wireguard-tools
│ xml-tooling-c
│ yelp-tools
│ zabbix
│ zsh
│ ==> Renamed Formulae
│ rebar@3 -> rebar3
│ Uninstalling fontconfig... (511 files, 3.2MB)
│ ==> Installing fontconfig from staticfloat/juliatranslated
│ ==> Downloading https://homebrew.bintray.com/bottles/fontconfig-2.13.0.sierra.bottle.tar.gz
│ Already downloaded: /Users/abradley/Library/Caches/Homebrew.jl/fontconfig-2.13.0.sierra.bottle.tar.gz
│ ==> Pouring fontconfig-2.13.0.sierra.bottle.tar.gz
│ ==> Regenerating font cache, this may take a while
│ ==> /Users/abradley/.julia/packages/Homebrew/l8kUw/deps/usr/Cellar/fontconfig/2.13.0/bin/fc-cache -frv
│ Last 15 lines from /Users/abradley/Library/Logs/Homebrew/fontconfig/post_install.01.fc-cache:
│ /usr/share/fonts: skipping, no such directory
│ /System/Library/Fonts: /System/Library/Fonts: failed to write cache
│ caching, new cache contents: 310 fonts, 0 dirs
│ /Library/Fonts: /Library/Fonts: failed to write cache
│ caching, new cache contents: 371 fonts, 0 dirs
│ /Users/abradley/Library/Fonts: /Users/abradley/Library/Fonts: failed to write cache
│ caching, new cache contents: 0 fonts, 0 dirs
│ /System/Library/Assets/com_apple_MobileAsset_Font3: /System/Library/Assets/com_apple_MobileAsset_Font3: failed to write cache
│ caching, new cache contents: 0 fonts, 0 dirs
│ /Users/abradley/.local/share/fonts: skipping, no such directory
│ /Users/abradley/.fonts: skipping, no such directory
│ /usr/local/var/cache/fontconfig: not cleaning unwritable cache directory
│ /Users/abradley/.cache/fontconfig: not cleaning unwritable cache directory
│ /Users/abradley/.fontconfig: not cleaning non-existent cache directory
│ /Users/abradley/.julia/packages/Homebrew/l8kUw/deps/usr/Cellar/fontconfig/2.13.0/bin/fc-cache: failed
│ Warning: The post-install step did not complete successfully
│ You can try again using `brew postinstall staticfloat/juliatranslated/fontconfig`
│ ==> Summary
│ 🍺  /Users/abradley/.julia/packages/Homebrew/l8kUw/deps/usr/Cellar/fontconfig/2.13.0: 511 files, 3.2MB
│ ERROR: LoadError: failed process: Process(`/Users/abradley/.julia/packages/Homebrew/l8kUw/deps/usr/bin/brew install --ignore-dependencies staticfloat/juliatranslated/fontconfig`, ProcessExited(1)) [1]
│ Stacktrace:
│  [1] error(::String, ::Base.Process, ::String, ::Int64, ::String) at ./error.jl:42
│  [2] pipeline_error at ./process.jl:695 [inlined]
│  [3] #run#505(::Bool, ::Function, ::Cmd) at ./process.jl:653
│  [4] run at ./process.jl:651 [inlined]
│  [5] #brew#4(::Bool, ::Bool, ::Bool, ::Bool, ::Bool, ::Function, ::Cmd) at /Users/abradley/.julia/packages/Homebrew/l8kUw/src/API.jl:19
│  [6] #brew at ./none:0 [inlined]
│  [7] #install_and_link#38(::Bool, ::Function, ::String) at /Users/abradley/.julia/packages/Homebrew/l8kUw/src/API.jl:492
│  [8] (::getfield(Homebrew, Symbol("#kw##install_and_link")))(::NamedTuple{(:verbose,),Tuple{Bool}}, ::typeof(Homebrew.install_and_link), ::String) at ./none:0
│  [9] macro expansion at ./logging.jl:310 [inlined]
│  [10] #add#29(::Bool, ::Bool, ::Function, ::String) at /Users/abradley/.julia/packages/Homebrew/l8kUw/src/API.jl:449
│  [11] #add at ./none:0 [inlined]
│  [12] #add#36(::Bool, ::Bool, ::Function, ::Homebrew.BrewPkg) at /Users/abradley/.julia/packages/Homebrew/l8kUw/src/API.jl:472
│  [13] #add at ./none:0 [inlined]
│  [14] #upgrade#8(::Bool, ::Function) at /Users/abradley/.julia/packages/Homebrew/l8kUw/src/API.jl:144
│  [15] #upgrade at ./none:0 [inlined]
│  [16] #update#6(::Bool, ::Function) at /Users/abradley/.julia/packages/Homebrew/l8kUw/src/API.jl:56
│  [17] update() at /Users/abradley/.julia/packages/Homebrew/l8kUw/src/API.jl:50
│  [18] top-level scope at /Users/abradley/.julia/packages/Homebrew/l8kUw/deps/build.jl:3
│  [19] include at ./boot.jl:317 [inlined]
│  [20] include_relative(::Module, ::String) at ./loading.jl:1038
│  [21] include(::Module, ::String) at ./sysimg.jl:29
│  [22] include(::String) at ./client.jl:388
│  [23] top-level scope at none:0
│ in expression starting at /Users/abradley/.julia/packages/Homebrew/l8kUw/deps/build.jl:1
│ 
│ Full log at /Users/abradley/.julia/packages/Homebrew/l8kUw/deps/build.log
└ @ Pkg.Operations /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.0/Pkg/src/Operations.jl:1068
  Building GSL ─────→ `~/.julia/packages/GSL/TcP5m/deps/build.log`
┌ Error: Error building `GSL`: 
│ ERROR: LoadError: UndefVarError: is_apple not defined
│ Stacktrace:
│  [1] top-level scope at none:0
│  [2] include at ./boot.jl:317 [inlined]
│  [3] include_relative(::Module, ::String) at ./loading.jl:1038
│  [4] include(::Module, ::String) at ./sysimg.jl:29
│  [5] include(::String) at ./client.jl:388
│  [6] top-level scope at none:0
│ in expression starting at /Users/abradley/.julia/packages/GSL/TcP5m/deps/build.jl:13
└ @ Pkg.Operations /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.0/Pkg/src/Operations.jl:1068
```

**Comment by simonbyrne** [comment]
Should now be fixed by #77/#78.
